### DISCUSSION TITLE
Two phase whitelist implementation approach question

### DISCUSSION THREAD

**Discussion by zhoukalex** [hashlips-lab/nft-erc721-collection#411] [Q&A]
Hi all, 

I want to have a two phase whitelist mint where 3500 gets minted in phase 1 and another 3500 gets minted in phase 2. I'm good with the default implementation of 1 TX per whitelist spot. 1 ETH address can only mint once in either phase 1 or 2. 

Whitelist can be flipped off/on, but what would be the best implementation of this 2 phase approach? Should we first begin by limiting maxSupply to 3500, and then creating a function to update it to 7000 once phase 2 begins? 

Thanks

```
const CollectionConfig: CollectionConfigInterface = {
  // ETH PROD  
  //testnet: Networks.hardhatLocal,
  // ETH testnetwork 
  testnet: Networks.ethereumTestnet,
  mainnet: Networks.ethereumMainnet,
  // The contract name can be updated using the following command:
  // yarn rename-contract NEW_CONTRACT_NAME
  // Please DO NOT change it manually!
  contractName: 'YourNftToken',
  tokenName: 'TokenTest',
  tokenSymbol: 'TBZ',
  hiddenMetadataUri: 'ipfs://test/hidden.json',
  maxSupply: 7000,
  whitelistSale: {
    price: 0.00005,
    maxMintAmountPerTx: 1,
  },
  preSale: {
    price: 0.00005,
    maxMintAmountPerTx: 5,
  },
  publicSale: {
    price: 0.00005,
    maxMintAmountPerTx: 5,
  },
  contractAddress: null,
  marketplaceIdentifier: 'my-nft-token',
  marketplaceConfig: Marketplaces.openSea,
  whitelistAddresses,
};
```

**Comment by liarco** [comment]
Hi @zhoukalex,
issues must be used for bug reports only. I'm moving this to the [Discussions section](https://github.com/hashlips-lab/nft-erc721-collection/discussions).

Thanks for your understanding.
